subscribers.js

```diff
@@ -73,6 +74,74 @@ const Subscribers = {
       return _ret(e);
     }
   },
+
+  async addTempUser(aEmail, aToken) {
+    if (!this.validateToken(aToken)) {
+      throw new Error("Invalid token, are you using Subscribers.generateToken?");
+    }
+
+    // Error code for attempting to add a duplicate entry on a UNIQUE key.
+    const DUPLICATE_ERROR = "23505";
+    try {
+      await dbq({
+        text: "INSERT INTO users_temp ( email, token ) VALUES ( $1, $2 );",
+        values: [ aEmail, aToken ],
+      });
+
+      return _ret();
+    } catch (e) {
+      if (e.code === DUPLICATE_ERROR) {
```

Comment:
Not sure how to test this, but `DUPLICATE_ERROR` above (L84) looks like a String, `const DUPLICATE_ERROR = "23505";`. Is `e.code` similarly a string version of a number? Since this is using `===` I don't think it will auto-convert to a number and this may not do what we expect.

Reply:
Yup, e.code is a string. Sigh. I'm actually glad I caught this, DUPLICATE_ERROR used to be a number so this check would never evaluate to true.

Reply:
If/when we convert Subscriber DB code to use sequelize, we may get duplicate error checking "for free".